src/runtime.js

```diff
@@ -408,36 +411,156 @@ function registerServiceForDoc(ampdoc, name, opt_ctor, opt_factory) {
+  const extensionIds = [];
+  if (doc.head) {
+    const parentLinks = {};
+    childElementsByTag(global.document.head, 'link').forEach(link => {
+      const href = link.getAttribute('href');
+      if (href) {
+        parentLinks[href] = true;
+      }
+    });
+
+    for (let n = doc.head.firstElementChild; n; n = n.nextElementSibling) {
+      const tagName = n.tagName;
+      const name = n.getAttribute('name');
+      const rel = n.getAttribute('rel');
+      if (n.tagName == 'TITLE') {
+        shadowRoot.AMP.title = n.textContent;
+        dev.fine(TAG, '- set title: ', shadowRoot.AMP.title);
+      } else if (tagName == 'META' && n.hasAttribute('charset')) {
+        // Ignore.
+      } else if (tagName == 'META' && name == 'viewport') {
+        // Ignore.
+      } else if (tagName == 'META') {
+        // TODO(dvoytenko): copy other meta tags.
+        dev.warn(TAG, 'meta ignored: ', n);
+      } else if (tagName == 'LINK' && rel == 'canonical') {
+        shadowRoot.AMP.canonicalUrl = n.getAttribute('href');
+        dev.fine(TAG, '- set canonical: ', shadowRoot.AMP.canonicalUrl);
+      } else if (tagName == 'LINK' && rel == 'stylesheet') {
+        // This must be a font definition: no other stylesheets are allowed.
+        const href = n.getAttribute('href');
+        if (parentLinks[href]) {
+          dev.fine(TAG, '- stylesheet already included: ', href);
+        } else {
+          parentLinks[href] = true;
+          const el = global.document.createElement('link');
+          el.setAttribute('rel', 'stylesheet');
+          el.setAttribute('type', 'text/css');
+          el.setAttribute('href', href);
+          global.document.head.appendChild(el);
+          dev.fine(TAG, '- import font to parent: ', href, el);
+        }
+      } else if (n.tagName == 'STYLE') {
+        if (n.hasAttribute('amp-boilerplate')) {
+          // Ignore.
+          dev.fine(TAG, '- ignore boilerplate style: ', n);
+        } else {
+          shadowRoot.appendChild(global.document.importNode(n, true));
+          dev.fine(TAG, '- import style: ', n);
+        }
+      } else if (n.tagName == 'SCRIPT' && n.hasAttribute('src')) {
+        dev.fine(TAG, '- src script: ', n);
+        const src = n.getAttribute('src');
+        const isRuntime = src.indexOf('/amp.js') != -1 ||
+            src.indexOf('/v0.js') != -1;
+        const customElement = n.getAttribute('custom-element');
+        const customTemplate = n.getAttribute('custom-template');
+        if (isRuntime) {
+          dev.fine(TAG, '- ignore runtime script: ', src);
+        } else if (customElement || customTemplate) {
+          // This is an extension.
+          extensions.loadExtension(customElement || customTemplate);
+          dev.fine(TAG, '- load extension: ', customElement || customTemplate);
+          if (customElement) {
+            extensionIds.push(customElement);
+          }
+        } else {
+          dev.warn(TAG, '- unknown script: ', n, src);
```

Comment:
I switched `user.error`. The reasoning: `cdn.ampproject.org` won't have any of these cases. But non-CDN documents will and thus this is a user error.